src/main/java/hudson/plugins/templateproject/ProxySCM.java

```diff
@@ -52,19 +47,12 @@ public String getProjectName() {
 		return projectName;
 	}
 
-	public String getExpandedProjectName(AbstractBuild<?, ?> build) {
-		return TemplateUtils.getExpandedProjectName(projectName, build);
-	}
-
-	// Primarily used for polling, not building.
-	public AbstractProject<?, ?> getProject() {
-		return TemplateUtils.getProject(projectName, null);
-	}
-
 	public SCM getProjectScm(AbstractBuild<?, ?> build) {
-		try {
-			return TemplateUtils.getProject(projectName, build).getScm();
-		} catch (Exception e) {
```

Comment:
NPE is just an example. IllegalStateException, IllegalArgumentException and other such guys are also theoretically possible. I have no idea what this <code>catch</code> clause expects to get, but I would avoid deleting it without prior investigation.